client/src/components/register.tsx

```diff
@@ -0,0 +1,139 @@
+import { useState, useEffect, useRef, FormEvent, ChangeEvent, KeyboardEvent } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import { IUserReg } from '../../../shared/interfaces/user.interface';
+
+const Register : React.FC = () => {
+	const navigate = useNavigate();
+	const emailField = useRef<HTMLInputElement>(null);
+	const passwordField = useRef<HTMLInputElement>(null);
+	const [user, setUser] = useState<IUserReg>({
+		firstName:"",
+		lastName:"",
+		email:"",
+		password:"",
+		confirmPassword:""
+	})
+	const [showPass, setShowPass] = useState(false)
+	const [capsCheck, setCapsCheck] = useState(false)
+	const [errors, setErrors] = useState<any>({
```

Comment:
it's not going to matter for us, but just wanted to make a note if we were to refactor we should properly type this. any defeats the point of typescript. i got told that often doing company take homes lol